### ISSUE TITLE
feat: Github Action with semantic versioning

### ISSUE THREAD

**Issue by mogwai** [pyannote/pyannote-audio#480]
<a href="https://github.com/mogwai"><img src="https://avatars1.githubusercontent.com/u/5440566?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [mogwai](https://github.com/mogwai)**
_Wednesday Oct 28, 2020 at 16:57 GMT_
_Originally opened as https://github.com/hbredin/pyannote-audio-v2/pull/52_

----

I found a pretty popular [package](https://github.com/semantic-release/semantic-release) that with a bit of tweaking does what we need. At the moment it will release on pushes or pull-request merges to `main`.

It needs commits to be formatted in a particular way, following the angualr commit guidelines: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines

As we are squash-merging, we could probably just do this through the naming of PR's. We may need something like this to help prevent bad commit messages but my question was how someone was supposed to revert this once they submit a PR.
https://github.com/conventional-changelog/commitlint

The things still missing are:

- Creation of docs which we don't have set up yet. 
- Publish to PyPi
  - What files to include in pypi

I tested this on a fork and confirmed that it works.

Closes #22

----

_**[mogwai](https://github.com/mogwai)** included the following code: https://github.com/hbredin/pyannote-audio-v2/pull/52/commits_

**Comment by mogwai** [comment]
<a href="https://github.com/mogwai"><img src="https://avatars1.githubusercontent.com/u/5440566?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mogwai](https://github.com/mogwai)**
_Saturday Oct 31, 2020 at 15:12 GMT_

----

I wasn't sure how to go forward with the docs and pypi. I've never published to pypi. Do you need some special access code? Should we just copy the action that is in v1?

Shall we leave the doc generation for a separate PR?

**Comment by mogwai** [comment]
<a href="https://github.com/hbredin"><img src="https://avatars2.githubusercontent.com/u/1080837?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [hbredin](https://github.com/hbredin)**
_Sunday Nov 01, 2020 at 21:26 GMT_

----

Before going further, we should have some kind of documentation on how this thing works and what the commit message convention should be. 

And yes, I'd rather have the PyPI deployment part of the PR as well.
And yes, documentation generation can wait for another PR.

**Comment by mogwai** [comment]
<a href="https://github.com/mogwai"><img src="https://avatars1.githubusercontent.com/u/5440566?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mogwai](https://github.com/mogwai)**
_Monday Nov 02, 2020 at 11:32 GMT_

----

Commit messages will be parsed by the semantic-versioning program following:


Commit message | Release type
-- | --
fix(pencil): stop graphite breaking when too much pressure applied | Patch Release
feat(pencil): add 'graphiteWidth' option | Minor Feature Release
perf(pencil): remove graphiteWidth optionBREAKING CHANGE: The graphiteWidth option has been removed.The default graphite width of 10mm is always used for performance reasons. | Major Breaking Release

If since the last versioned (tagged) commit, there has been the following commits and messages:

```
feat: Added Hydra Network Segmentation example
fix: Random number generator issue resolve #54 
```

It would bundle the commits to together as a minor feature release as this is the highest rank of change in the set of commits. 

Essentially we need to be merging in PR's with names and messages corresponding to the angular commit message guidelines although I think we can just get away with making sure the PR titles are tagged with one of the following types: 

https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#type

**Comment by mogwai** [comment]
<a href="https://github.com/mogwai"><img src="https://avatars1.githubusercontent.com/u/5440566?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mogwai](https://github.com/mogwai)**
_Monday Nov 02, 2020 at 13:03 GMT_

----

I'd definitely recommend double checking this works in a fork and testing yourself it does what we want. It's based on the git tags essentially.
